BusBoard.Web/ViewModels/BusInfo.cs

```diff
@@ -1,13 +1,45 @@
-﻿namespace BusBoard.Web.ViewModels
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using BusBoard.Api.Postcode;
+using BusBoard.Api.Tfl;
+using BusBoard.Api.Tfl.Responses;
+
+namespace BusBoard.Web.ViewModels
 {
   public class BusInfo
   {
+    public IEnumerable<StopPoint> StopPoints { get; } = new List<StopPoint>();
+    public string ErrorMessage { get; } = "";
+
+    private TflApi tflApi = new TflApi();
+    private PostcodeApi postcodeApi = new PostcodeApi();
+
     public BusInfo(string postCode)
     {
-      PostCode = postCode;
+      try
+      {
+        var postcodeData = postcodeApi.GetPostcodeData(postCode);
+
+        if (postcodeData == null)
+        {
+          return;
```

Comment:
In what scenario would this be `null`? 🙂

Reply:
This is `null` when the user clicks "Get Bus Info" without entering any text into the text box. I probably should check for this and return a "Postcode not found error" though

Reply:
Changed in 4be48dd 🙂

Reply:
It might be worth throwing an exception in that case as well, to make the error handling more consistent? 🙂

Reply:
Ah sorry, I see you already have done that 🙂

Reply:
No worries! Yeah it feels like a bug in the API implementation, it should be telling you your postcode is invalid with an exception instead of quietly returning a `null` if you pass it a `null` 🙂